src/Core/Result.php

```diff
@@ -85,38 +84,50 @@ public function __destruct()
      */
     final public function resolve(?float $timeout = null): bool
     {
-        if ($this->isResolved) {
+        if (null !== $this->resolveResult) {
+            if ($this->resolveResult instanceof \Exception) {
+                throw $this->resolveResult;
+            }
+
+            if (\is_bool($this->resolveResult)) {
+                return $this->resolveResult;
+            }
+        }
```

Comment:
Maybe throw exception if it is unexpected value.

Reply:
done